### PR TITLE
chore: improve tinygo maintenance path

### PR DESCRIPTION
**Pull request by otjdiepluong**
Summary:
- Add or tighten focused edge-case tests or type assertions in main_test.go, corpus_test.go related to CI, Docker, Kubernetes, build tooling, release automation; avoid docs-only changes and broad refactors.
- Keep the change narrow so it is straightforward to review.

Notes:
- I kept this scoped to the relevant implementation and tests.

### CODE REVIEW

File: corpus_test.go [hunk 1]
```diff
@@ -150,6 +150,63 @@ func TestCorpus(t *testing.T) {
 	}
 }
 
+func TestLoadRepos(t *testing.T) {
```
Comment (aykevl) [inline corpus_test.go:153]: Can you add a comment here explaining what this test tests for?

**Comment by deadprogram** [comment]
Closing due to inactivity. Please reopen if still needed. Thank you.
